client/network/src/discovery.rs

```diff
@@ -298,6 +314,23 @@ impl DiscoveryBehaviour {
 			(id, size)
 		})
 	}
+
+	/// Can the given `Multiaddr` be put into the DHT?
+	///
+	/// This test is successful only for global IP addresses and DNS names.
+	//
+	// NB: Currently all DNS names are allowed and no check for TLD suffixes is done
+	// because the set of valid domains is highly dynamic and would require frequent
+	// updates, for example by utilising publicsuffix.org or IANA.
+	pub fn can_add_to_dht(&self, addr: &Multiaddr) -> bool {
+		let ip = match addr.iter().next() {
+			Some(Protocol::Ip4(ip)) => IpNetwork::from(ip),
```

Comment:
I guess once the `std` API stabilizes we can use https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#method.is_global.

Reply:
Yes, I pulled in `ip_network` precisely for this reason.